### PR TITLE
Initial pull request

### PR DESCRIPTION
**Pull request by PHronek**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+language: ruby
+rvm:
+  - 2.2.3
```
Comment (boblin) [inline .travis.yml]: Have you consider add latest versions like '2.2' and '2.3'? Also to add 'ruby-head' to know that is compatible with the latest ruby?

File: Gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,59 @@
+PATH
```
Comment (boblin) [inline Gemfile.lock]: `Gemfile.lock' should not be part of git in case of gem.

File: bin/setup [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (boblin) [inline bin/setup]: Is this file really needed?

File: ecb_rates.gemspec [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+require 'ecb_rates/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = "ecb_rates"
+  spec.version       = EcbRates::VERSION
+  spec.authors       = ["Pavel Hronek"]
+  spec.email         = ["hronek@uol.cz"]
+
+  spec.summary       = %q{Gem for downloading currency exchange rates}
+  spec.description   = %q{Gem for downloading currency exchange rates}
```
Comment (boblin) [inline ecb_rates.gemspec]: What about info about source of rates?

File: README.md [hunk 5]
```diff
@@ -1 +1,23 @@
-# ecb_rates
+# Usage:
```
Comment (boblin) [inline README.md]: I think I would be helpful to add information about source of exchange rates like website url.

File: ecb_rates.gemspec [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  # delete this section to allow pushing this gem to any host.
+  if spec.respond_to?(:metadata)
+    spec.metadata['allowed_push_host'] = "TODO: Set to 'http://mygemserver.com'"
+  else
+    raise "RubyGems 2.0 or newer is required to protect against public gem pushes."
+  end
+
+  spec.files         = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+  spec.bindir        = "exe"
+  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
```
Comment (boblin) [inline ecb_rates.gemspec]: I think there is no `exe` here.

File: spec/fixtures/eurofxref-daily.xml [hunk 7]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (boblin) [inline spec/fixtures/eurofxref-daily.xml]: Where is this file from? Is it manually downloaded? How can be updated in future?

Have you consider to use [vcr](https://github.com/vcr/vcr) for automated downloading and updating source files?

File: README.md [hunk 8]
```diff
@@ -1 +1,23 @@
...
+
```
Comment (boblin) [inline README.md]: Installation part would be helpful.

**Comment by boblin** [comment]
Have you consider to use rubocop together with overcommit here? If yes, rubocop and overcommit config files are needed.

File: spec/fixtures/eurofxref-daily.xml (hunk 7 above)
Comment (PHronek) [inline spec/fixtures/eurofxref-daily.xml]: Yeah, they are manually downloaded from the source. I thought that it is easier to simply download static xml files and use them in tests because if their format or structure changes we will need to alter code anyway.

Comment (boblin) [inline spec/fixtures/eurofxref-daily.xml]: `vcr` is just helper but not necessary. Anyway we should probably add comment with source url where is this file from.

File: .travis.yml (hunk 1 above)
Comment (Masa331) [inline .travis.yml]: ruby 2.3 is already final?

File: README.md [hunk 9]
```diff
@@ -1 +1,47 @@
-# ecb_rates
+
+
+# EcbRates:
+
+Simple gem for getting exchange rates between from EUR to various other currencies.
```
Comment (Masa331) [inline README.md]: Describe that rates are taken from ECB

File: ecb_rates.gemspec [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+require 'ecb_rates/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = 'ecb_rates'
+  spec.version       = EcbRates::VERSION
+  spec.authors       = ['Pavel Hronek']
+  spec.email         = ['hronek@uol.cz']
+
+  spec.summary       = %q{Gem for downloading currency exchange rates}
+  spec.description   = %q{Gem for downloading currency exchange rates}
```
Comment (Masa331) [inline ecb_rates.gemspec]: form ECB

File: ecb_rates.gemspec [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  spec.homepage      = "TODO: Put your gem's website or public repo URL here."
```
Comment (Masa331) [inline ecb_rates.gemspec]: url to github repo

File: .travis.yml (hunk 1 above)
Comment (boblin) [inline .travis.yml]: My mistake. I meant ruby '2.1' (our current version) and '2.2'.

File: ecb_rates.gemspec [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+  # Prevent pushing this gem to RubyGems.org by setting 'allowed_push_host', or
+  # delete this section to allow pushing this gem to any host.
+  if spec.respond_to?(:metadata)
+    spec.metadata['allowed_push_host'] = "TODO: Set to 'http://mygemserver.com'"
+  else
+    raise 'RubyGems 2.0 or newer is required to protect against public gem pushes.'
+  end
+
+  spec.files         = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+  spec.bindir        = 'exe'
```
Comment (boblin) [inline ecb_rates.gemspec]: This line is probably irrelevant.

File: lib/ecb_rates/application.rb [hunk 13]
```diff
@@ -0,0 +1,28 @@
+require 'date'
+
+module EcbRates
+  class Application
```
Comment (boblin) [inline lib/ecb_rates/application.rb]: Have you consider to move content of `class Application` right to module EcbRates like [here](https://github.com/jnicklas/capybara/blob/master/lib/capybara.rb)?

Comment (PHronek) [inline lib/ecb_rates/application.rb]: I think its great for usage simplificatiod and i added simple module method for that case. Otherwise  think its better to have content separated.  But moving all the errors to EcbRates root module might be also a good idea.

Comment (boblin) [inline lib/ecb_rates/application.rb]: Please, could you describe what it simplify? In my opinion usage looks slightly complicated. Like:

```ruby
raise_error EcbRates::Application::CurrencyNotSupported
# vs
raise_error EcbRates::CurrencyNotSupported
```

Comment (PHronek) [inline lib/ecb_rates/application.rb]: I mean this part:

```ruby
EcbRates.exchange_rate('JPY')
```
instead of:

```ruby
app = EcbRates::Application.new
app.exchange_rates('JPY')
```

and yeah youre right, exceptions should be moved to EcbRates module.
